### PR TITLE
Run `gn check` after lint to find problems with gn files

### PR DESCRIPTION
**Pull request by bsclifton**
Per convo here:
https://bravesoftware.slack.com/archives/C7VLGSR55/p1569944801254700

Let's see how CI does with this (don't merge please!)

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -992,6 +996,12 @@ def lint() {
     """
 }
 
+def gn_check() {
+    sh """
+        gn check ${OUT_DIR}
```
Comment (mihaiplesa) [inline Jenkinsfile]: isn’t build dir available only after build?

Comment (mihaiplesa) [reply Jenkinsfile]: do we want this added to Windows as well?

Comment (iefremov) [reply Jenkinsfile]: sort of - the dir is generated after `gn gen out/Release`. We don't have to compile everything to do `gn check`

Comment (bsclifton) [reply Jenkinsfile]: @iefremov would the directory exist after `npm run init`? I don't think so, right? Is it safe to make an empty directory? or does it expect content to be there. Confused why it needs the directory if compilation not needed

Comment (iefremov) [reply Jenkinsfile]: sorry for the confusion. The checker needs your "setup" to be configured - i.e. all variables in gn files to be set correctly. To do this, you have to run `gn gen out/<...>` for the build configuration that you want. I suppose that in our brave world `gn gen` happens as part of `npm run build`

**Comment by bsclifton** [comment]
Will want to revisit this... (don't have enough time to follow through on everything yet)

**Pull request closed without merging**